Response code range not supported

Thanks for adding check of response code!

Response codes may contain a range of code in the format XXX
https://swagger.io/specification/#responsesObject

My output
```
        if errors:
>           raise AssertionError(*errors)
E           AssertionError: Received a response with a status code, which is not defined in the schema: 404
E           
E           Declared status codes: 4XX, 200
```

--- a/docs/changelog.rst
+++ b/docs/changelog.rst
@@ -6,6 +6,11 @@ Changelog
 `Unreleased`_
 -------------
 
+Added
+~~~~~
+
+- Support for response code ranges. `#330`_
+
 `0.20.0`_ - 2019-12-12
 ----------------------
 
@@ -500,6 +505,7 @@ Fixed
 .. _0.3.0: https://github.com/kiwicom/schemathesis/compare/v0.2.0...v0.3.0
 .. _0.2.0: https://github.com/kiwicom/schemathesis/compare/v0.1.0...v0.2.0
 
+.. _#330: https://github.com/kiwicom/schemathesis/issues/330
 .. _#322: https://github.com/kiwicom/schemathesis/issues/322
 .. _#319: https://github.com/kiwicom/schemathesis/issues/319
 .. _#315: https://github.com/kiwicom/schemathesis/issues/315

--- a/src/schemathesis/checks.py
+++ b/src/schemathesis/checks.py
@@ -1,4 +1,6 @@
-from typing import TYPE_CHECKING, Callable, Tuple, Union
+import string
+from itertools import product
+from typing import TYPE_CHECKING, Any, Callable, Dict, Generator, Tuple, Union
 
 import jsonschema
 import requests
@@ -22,13 +24,20 @@ def status_code_conformance(response: GenericResponse, case: "Case") -> None:
     # "default" can be used as the default response object for all HTTP codes that are not covered individually
     if "default" in responses:
         return
-    allowed_response_statuses = list(map(str, responses))
+    allowed_response_statuses = list(_expand_responses(responses))
     if str(response.status_code) not in allowed_response_statuses:
         message = (
             f"Received a response with a status code, which is not defined in the schema: "
             f"{response.status_code}\n\nDeclared status codes: {', '.join(allowed_response_statuses)}"
         )
         raise AssertionError(message)
+
+
+def _expand_responses(responses: Dict[Union[str, int], Any]) -> Generator[str, None, None]:
+    for code in responses:
+        chars = [list(string.digits) if char == "X" else [char] for char in str(code).upper()]
+        for expanded in product(*chars):
+            yield "".join(expanded)
 
 
 def content_type_conformance(response: GenericResponse, case: "Case") -> None:

--- a/test/runner/test_checks.py
+++ b/test/runner/test_checks.py
@@ -4,7 +4,7 @@ import pytest
 import requests
 
 from schemathesis import models
-from schemathesis.checks import content_type_conformance, response_schema_conformance
+from schemathesis.checks import content_type_conformance, response_schema_conformance, status_code_conformance
 from schemathesis.schemas import BaseSchema
 
 
@@ -42,6 +42,23 @@ def case(request, swagger_20) -> models.Case:
 )
 def test_content_type_conformance_valid(response, case):
     assert content_type_conformance(response, case) is None
+
+
+@pytest.mark.parametrize("value", (400, 405))
+def test_status_code_conformance_valid(value, swagger_20):
+    response = make_response()
+    response.status_code = value
+    case = make_case(swagger_20, {"responses": {"4XX"}})
+    status_code_conformance(response, case)
+
+
+@pytest.mark.parametrize("value", (400, 405))
+def test_status_code_conformance_invalid(value, swagger_20):
+    response = make_response()
+    response.status_code = value
+    case = make_case(swagger_20, {"responses": {"5XX"}})
+    with pytest.raises(AssertionError):
+        status_code_conformance(response, case)
 
 
 @pytest.mark.parametrize(

